pkg/segment/aggregations/segaggs.go

```diff
@@ -141,7 +141,7 @@ func PostQueryBucketCleaning(nodeResult *structs.NodeResult, post *structs.Query
 		if len(nodeResult.TransactionEventRecords) > 0 {
 			nodeResult.NextQueryAgg = agg
 			return nodeResult
-		} else if nodeResult.PerformAggsOnRecs && len(recs) > 0 {
+		} else if nodeResult.PerformAggsOnRecs && recs != nil {
 			nodeResult.NextQueryAgg = agg
 			return nodeResult
 		}
```

Comment:
Just curious: does this change actually change any behavior? I thought Go implements `len()` so that `len(nil) == 0`, so the only potential difference is when `recs` is non-nil but has zero items, but you're not using `recs` inside the block. So it looks like nothing changes; or am I missing something?